Sources/MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -149,6 +149,8 @@ open class RouteProgress: Codable {
                                               spokenInstructionIndex: currentLegProgress.currentStepProgress.spokenInstructionIndex)
         calculateLegsCongestion()
         updateDistanceTraveled(with: location)
+        updateDistanceToIntersection(from: location)
+        updateIntersectionIndex()
```

Comment:
Maybe we could pass `intersectionIndex` from `currentLegProgress.currentStepProgress` through `RouteLegProgress` and then `RouteStepProgress` recreation?
I'd prefer relying on the data coming from `NavigationStatus` (which doesn't seem obsolete from the last update) to logic from `LegacyRouteController`.

This would also update `intersectionIndex` before relying on it in `updateDistanceToIntersection`.

Reply:
Yes, the `RouteProgress.updateIntersectionIndex()` is only called once when the route refresh happened. Except this one, the `intersectionIndex` is still handled by the `NavigationStatus`. The `RouteController` only called the `updateDistanceToIntersection(from:)` like it previously did. No more change to the `RouteController`  behavior.

Reply:
We now have a problem with `intersectionIndex` being lost when [recreating `RouteStepProgress`](https://github.com/mapbox/mapbox-navigation-ios/blob/main/Sources/MapboxCoreNavigation/RouteLegProgress.swift#L198), thus filled with the default value. We have that info from previous step progress, like we do for [`spokenInstructionIndex`](https://github.com/mapbox/mapbox-navigation-ios/blob/main/Sources/MapboxCoreNavigation/RouteProgress.swift#L162). My proposal is to pass this data to the new `RouteStepProgress` instead of calculating it manually.

Reply:
The other thing here is that in `updateDistanceToIntersection` we already rely on `intersectionIndex` being up-to-date [here in the `upcomingIntersection`](https://github.com/mapbox/mapbox-navigation-ios/pull/4193/files#diff-146794cc23a9983b907b9c9f3a8805421e7a92ca942b338f19fca699ad7f93b6R207). So we need to have it updated before calling `updateDistanceToIntersection`. If we use `intersectionIndex` from the previous `RouteStepProgress` it should have the current value.

Reply:
I see. I removed the manually calculating part and used the passed down `intersectionIndex` as the value to update the following intersection related properties.